addOns/custompayloads/src/test/java/org/zaproxy/zap/extension/custompayloads/CustomPayloadsParamUnitTest.java

```diff
@@ -0,0 +1,140 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.custompayloads;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import org.apache.commons.configuration.HierarchicalConfiguration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.zaproxy.zap.utils.ZapXmlConfiguration;
+
+/** Unit test for {@link CustomPayloadsParam}. */
+class CustomPayloadsParamUnitTest {
+
+    private CustomPayloadsParam param;
+    private ZapXmlConfiguration configuration;
+
+    @BeforeEach
+    void setUp() {
+        param = new CustomPayloadsParam();
+        configuration = new ZapXmlConfiguration();
+    }
+
+    @Test
+    void shouldHaveConfigVersionKey() {
+        // Given / When / Then
+        assertThat(param.getConfigVersionKey(), is(equalTo("custompayloads[@version]")));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"true", "false"})
+    void shouldLoadConfirmRemoveFromConfig(boolean state) {
+        // Given
+        configuration.setProperty(CustomPayloadsParam.CONFIRM_REMOVE_PAYLOAD_KEY, state);
+        // When
+        param.load(configuration);
+        // Then
+        assertThat(param.isConfirmRemoveToken(), is(equalTo(state)));
+    }
+
+    @Test
+    void shouldDefaultConfirmRemoveTrue() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.isConfirmRemoveToken(), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldNotHaveNextPayloadIdOnUpdateFromUnversioned() {
+        // Given
+        String configKey = "custompayloads.nextPayloadId";
+        configuration = new ZapXmlConfiguration();
+        configuration.setProperty(configKey, 72);
+        param.load(configuration);
+        // When / Then
+        assertThrows(NoSuchElementException.class, () -> configuration.getInt(configKey));
+    }
+
+    @Test
+    void shouldRemoveIdsFromCustomPayloadsOnUpdate() {
+        // Given
+        String configKey = "custompayloads.categories.category(0).payloads.payload(0).payload.id";
+        createTestConfig(configuration);
+        // Before update there should be a payload with ID == 0
+        assertThat(configuration.getInt(configKey), is(equalTo(0)));
+        // When
+        param.updateConfigsImpl(-1);
+        // Then
+        assertThrows(NoSuchElementException.class, () -> configuration.getInt(configKey));
```

Comment:
I've addressed a bunch of this. I can't pinpoint where I've gone wrong with the loading, and the payload data not actually being there.

Reply:
I've pushed the latest, the tests will fail and there's still printlns to remove. Could you point out what I've overlooked?

Reply:
The `updateConfigsImpl` needs to be tweaked, instead of the save to call (which clears the configs) it should just clear the IDs (e.g. `getConfig().clearProperty("custompayloads.categories.category.payloads.payload.id");`).